### PR TITLE
TASK-1107, TASK-1150, Changes for input 'ref_path_to_set' and 'ref_path' output

### PR DESCRIPTION
**Pull request by ugswork**
Changes include the following:

TASK-1150
- The way object selector is created to get objects using 'ref_path_to_set' input parameter.
TASK-1107
- Addition of input parameter 'include_set_item_ref_paths' to all SetAPI get functions.
  include_set_item_ref_paths - 1 or 0, if 1, additionally provides ref_path for each item
                                     in the set. The ref_path returned for each item is either
                                         ref_path_to_set;item_ref  (if ref_path_to_set is given) or
                                         set_ref;item_ref  (if ref_path_to_set is not given)
- Addition of tests for this additional parameter in all SetAPI get functions

### CODE REVIEW

**Comment by briehl** [comment]
Just noting here that tests are failing because there isn't a proper .travis.yml file set up. It might be worth adding one to do kb-sdk validate (essentially a no-op) if nothing else.

File: test/ReadsAlignmentSet_basic_test.py [hunk 1]
```diff
@@ -282,6 +283,92 @@ def test_get_alignment_set(self):
...
+        self.assertEquals(alignment_set_ref, info_to_ref(fetched_set["info"]))
+        for item in fetched_set["data"]["items"]:
+            self.assertNotIn("info", item)
+            self.assertIn("ref", item)
+            self.assertIn("label", item)
+            self.assertIn("ref_path", item)
+            self.assertEquals(item["ref_path"], alignment_set_ref + ";" + item["ref"])
+
+    # Following test uses object refs from a narrative. Comment the next line to run the test
+    @unittest.skip("skipped test_get_rnaseq_alignment_set_ref_path")
```
Comment (briehl) [inline test/ReadsAlignmentSet_basic_test.py]: It's worth taking the time to make dummy objects (not reliant on existing workspaces) so anyone can run tests here. There should be enough tooling available in test/util.py

File: test/ReadsAlignmentSet_basic_test.py [hunk 2]
```diff
@@ -282,6 +283,92 @@ def test_get_alignment_set(self):
...
+        pprint(fetched_set_with_ref_path)
+        print("==========================")
+
+    # Following test uses object refs from a narrative. Comment the next line to run the test
+    @unittest.skip("skipped test_get_kbasesets_alignment_set_ref_path")
+    def test_get_kbasesets_alignment_set_ref_path(self):
+        """
+        appdev KBasesets.AlignmentSet ref
+        """
+        alignment_set_ref = '5264/36/10'
```
Comment (briehl) [inline test/ReadsAlignmentSet_basic_test.py]: As above, it's worth making a dummy alignment set. In fact, there should be one made in the setUpClass function.

**Review by briehl — CHANGES_REQUESTED** [review]

File: test/ReadsAlignmentSet_basic_test.py (hunk 1 above)
Comment (ugswork) [reply test/ReadsAlignmentSet_basic_test.py]: Dummy objects are created for every set type, and there are tests which make use of the dummy objects. The tests with existing workspace objects are in addition to those, which i used for initial testing. I have removed them.

**Review by briehl — APPROVED** [review]

**Pull request merged**